### PR TITLE
refactor: Use swr for refresh hooks that doesn't dispatch to redux

### PR DESCRIPTION
**Pull request by memoyil**

### CODE REVIEW

File: src/views/FarmAuction/hooks/useCurrentFarmAuction.ts [hunk 1]
```diff
@@ -1,58 +1,41 @@
...
+    async () => {
+      const auctionId = await farmAuctionContract.currentAuctionId()
+      const auctionData = await farmAuctionContract.auctions(auctionId)
+      return processAuctionData(auctionId.toNumber(), auctionData)
+    },
+    { refreshInterval: FAST_INTERVAL },
+  )
+
+  const { data: bidders = null, mutate: refreshBidders } = useSWR(
+    ['farmAuction', 'bidders'],
```
Comment (chef-jojo) [inline src/views/FarmAuction/hooks/useCurrentFarmAuction.ts:24]: check `currentAuction ? ['farmAuction', 'bidders'] : null` in cause currentAuction maybe null at first

Comment (memoyil) [reply src/views/FarmAuction/hooks/useCurrentFarmAuction.ts:24]: I will fix that

File: src/views/Home/components/WinSection/LotteryCardContent.tsx [hunk 2]
```diff
@@ -26,8 +27,17 @@ const LotteryCardContent = () => {
   const { observerRef, isIntersecting } = useIntersectionObserver()
   const [loadData, setLoadData] = useState(false)
   const [lotteryId, setLotteryId] = useState<string>(null)
-  const [currentLotteryPrize, setCurrentLotteryPrize] = useState<BigNumber>(null)
   const cakePriceBusdAsString = usePriceCakeBusd().toString()
+  const { data: currentLotteryPrize = null } = useSWR(
+    lotteryId ? ['lottery', 'currentLotteryPrize', cakePriceBusdAsString] : null,
```
Comment (chef-jojo) [inline src/views/Home/components/WinSection/LotteryCardContent.tsx:32]: cake price changed a lot, maybe not worth to add this into key

Comment (memoyil) [reply src/views/Home/components/WinSection/LotteryCardContent.tsx]: Done

File: src/views/FarmAuction/hooks/useCurrentFarmAuction.ts (hunk 1 above)
Comment (memoyil) [reply src/views/FarmAuction/hooks/useCurrentFarmAuction.ts]: Done

**Review by chefilip — COMMENTED** [review]
Thanks for your contribution ! Great work !

File: src/views/Home/components/WinSection/PredictionCardContent.tsx [hunk 3]
```diff
@@ -19,7 +20,9 @@ const PredictionCardContent = () => {
   const { observerRef, isIntersecting } = useIntersectionObserver()
   const [loadData, setLoadData] = useState(false)
   const bnbBusdPrice = useBNBBusdPrice()
-  const [bnbWon, setBnbWon] = useState(0)
+  const { data: bnbWon = 0 } = useSWR(loadData ? ['prediction', 'bnbWon'] : null, async () => getTotalWon(), {
```
[the remark this answers is not in the document]
Comment (memoyil) [reply src/views/Home/components/WinSection/PredictionCardContent.tsx]: Done

**Review by chef-jojo — APPROVED** [review]

**Pull request merged**